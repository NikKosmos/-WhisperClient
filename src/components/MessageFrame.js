import React from 'react'
import PropTypes from 'prop-types'
import { MessageTimestamp } from './MessageTimestamp'

function MessageFrame(props) {

    const { message } = props

    return (
        <div>
            <div>
                <p>{ message.authorName }</p>
            </div>
            <div>
                <p>{ message.authorEmail }</p>
            </div>
            <div>
                <p>{ message.text }</p>
            </div>
            <MessageTimestamp time = { message.time } />
        </div>
    )
}

MessageFrame.propTypes = {
    message: PropTypes.exact({
        _id: PropTypes.string,
        chatId: PropTypes.string.isRequired,
        authorName: PropTypes.string.isRequired,
        authorEmail: PropTypes.string.isRequired,
        time: PropTypes.oneOfType([PropTypes.number, PropTypes.string, PropTypes.instanceOf(Date)]).isRequired,
        text: PropTypes.string.isRequired,
        wasMessageReceived: PropTypes.bool,
    })
}

export { MessageFrame }