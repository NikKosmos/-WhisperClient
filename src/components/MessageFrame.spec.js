import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json';

import { MessageFrame } from './MessageFrame'

describe('MessageFrame', () => {
    const mockMessage = {
        _id: '0',
        chatId: '0',
        authorName: 'captain Nemo',
        authorEmail: 'nemo@mail.ru',
        time: '2019-12-25T16:40:12.526Z',
        text: 'some message',
        wasMessageReceived: true,
    }

    it ('should render correctly', () => {
        const testComponent = shallow(<MessageFrame 
            message = { mockMessage }
        />)

        expect(shallowToJson(testComponent)).toMatchSnapshot();
      })

})