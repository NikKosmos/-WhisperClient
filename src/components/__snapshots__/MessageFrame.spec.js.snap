// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageFrame should render correctly 1`] = `
<div>
  <div>
    <p>
      captain Nemo
    </p>
  </div>
  <div>
    <p>
      nemo@mail.ru
    </p>
  </div>
  <div>
    <p>
      some message
    </p>
  </div>
  <MessageTimestamp
    time="2019-12-25T16:40:12.526Z"
  />
</div>
`;
